docs/src/docs/asciidoc/guides/couchbase.adoc

```diff
@@ -0,0 +1,133 @@
+= Spring Session with Couchbase
+Mariusz Kopylec
+:toc:
+
+This guide describes how to use Spring Session backed by Couchbase.
+
+== Updating Dependencies
+Before you use Spring Session, you must ensure to update your dependencies.
+We assume you are working with a working Spring Boot web application.
+If you are using Gradle, ensure to add the following dependencies:
+
+.gradle.build
+[source,groovy]
+[subs="verbatim,attributes"]
+----
+dependencies {
+    compile group: 'org.springframework.data', name: 'spring-data-couchbase', version: '+'
+    compile group: 'org.springframework.boot', name: 'spring-boot-starter-web', version: '+'
+}
+----
+
+ifeval::["{version-snapshot}" == "true"]
+Since We are using a SNAPSHOT version, we need to ensure to add the Spring Snapshot Maven Repository.
+Ensure you have the following in your build.gradle:
+
+.gradle.build
+[source,groovy]
+----
+repositories {
+    maven { url 'https://repo.spring.io/libs-snapshot' }
+}
+----
+endif::[]
+
+ifeval::["{version-milestone}" == "true"]
+Since We are using a Milestone version, we need to ensure to add the Spring Milestone Maven Repository.
+Ensure you have the following in your build.gradle:
+
+.gradle.build
+[source,groovy]
+----
+repositories {
+    maven { url 'https://repo.spring.io/libs-milestone' }
+}
+----
+endif::[]
+
+[[how-to-use]]
+== How to use
+
+**Basic usage:**
+
+Configure Couchbase connection using Spring Data Couchbase. You can find a minimal configuration example link:http://projects.spring.io/spring-data-couchbase/#quick-start[here].
+
+Enable Couchbase backed HTTP session using `@EnableCouchbaseHttpSession`:
+
+[source,java]
+----
+@EnableCouchbaseHttpSession(namespace = "my-application")
+@SpringBootApplication
+public class Application {
+
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+}
+----
+
+Simply use `HttpSession` interface to control HTTP session. For example:
+
+[source,java]
+----
+@Controller
+public class SessionController {
+
+    @RequestMapping("uri")
+    public void doSomething(HttpSession session) {
+        ...
+    }
+}
+----
+
+**Namespaces support:**
```

Comment:
If we are going to consider global namespace, it will need to be a distinct ticket and PR. We don't want to change too much in one ticket / commit as this makes troubleshooting difficult.